core/src/zeit/content/volume/volume.py

```diff
@@ -228,6 +233,32 @@ def _is_valid_cover_id_and_product_id(self, cover_id, product_id):
         product_ids = [prod.id for prod in self._all_products]
         return cover_id in cover_ids and product_id in product_ids
 
+    def all_content_via_storage(self, additional_query_constraints=None):
+        """
+        Get all content for this volume via storage
+        If u pass a list of additional query clauses, they will be added as
+        an AND-operand to the query.
+        """
+        if not additional_query_constraints:
+            additional_query_constraints = ''
+        products = [x.id for x in self._all_products]
+        query = f"""
+        volume_year = :year
+        AND volume_number = :volume
+        AND product IN :products
+        NOT channels @> '[["zeit-magazin", "wochenmarkt"]]'
+        {additional_query_constraints}
+        """
+        query = select(ConnectorModel).where(
+            sql(query).bindparams(
+                bindparam(key='products', value=products, expanding=True),
```

Comment:
```suggestion
                products=bindparam(None, products, expanding=True),
```

find ich immernoch klunky, aber etwas _weniger_ klunky

Reply:
man könnte jetzt statt zwei Strings mit Queries zusammen zubauen, sowas mit [Subquery](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.subquery) bauen. Was hälst du davon?

Reply:
ok, lassen wir ✅

Reply:
der Vorschlag verpasst mir einen CompileError. 
```
sqlalchemy.exc.CompileError: No literal value renderer is available for literal value "BindParameter('%(4557873440 param)s', ['ZEI', 'ZMLB'], type_=String())" with datatype NULL
```

Reply:
bin nicht optimistisch, weil [siehe](https://github.com/sqlalchemy/sqlalchemy/discussions/9006)